### PR TITLE
fix (api): Fix bad verification when an admin has access group

### PR DESCRIPTION
**Pull request by callapa**
## Description

When a contact is an admin we checked these access group when it is not necessary.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 1]
```diff
@@ -848,7 +851,9 @@ public function findServiceGroups(): array
             }
         }
 
-        $request ='SELECT SQL_CALC_FOUND_ROWS DISTINCT sg.* FROM `:dbstg`.`servicegroups` sg ' . $subRequest;
+        $request =
+            'SELECT SQL_CALC_FOUND_ROWS DISTINCT sg.* 
+            FROM `:dbstg`.`servicegroups` sg ' . $subRequest;
```
Comment (sc979) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]: are the back quote around the bound dbstg mandatory ?

File: src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php [hunk 2]
```diff
@@ -126,14 +126,14 @@ public function findLatestHostAcknowledgement(int $hostId): ?Acknowledgement
...
+        $accessGroupFilter = $this->isAdmin()
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = ack2.host_id
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php [hunk 3]
```diff
@@ -172,15 +172,15 @@ public function findLatestServiceAcknowledgement(int $hostId, int $serviceId): ?
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = ack2.host_id
                   AND acl.service_id = ack2.service_id
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php [hunk 4]
```diff
@@ -227,15 +227,15 @@ private function findLatestAcknowledgementOf(int $type = self::TYPE_HOST_ACKNOWL
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = ack2.host_id'
                 .  (($type === self::TYPE_SERVICE_ACKNOWLEDGEMENT) ? ' AND acl.service_id = ack2.service_id ' : '')
                 . ' INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Acknowledgement/AcknowledgmentRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 5]
```diff
@@ -123,15 +123,15 @@ public function findHosts(): array
...
+            ? ''
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = h.host_id
                   AND acl.service_id IS NULL
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 6]
```diff
@@ -205,15 +205,15 @@ public function findHostsByHostsGroups(array $hostsGroupsIds): array
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = h.host_id
                   AND acl.service_id IS NULL
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 7]
```diff
@@ -261,15 +261,15 @@ public function findHostsByServiceGroups(array $servicesGroupsIds): array
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = h.host_id
                   AND acl.service_id = srv.service_id
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 8]
```diff
@@ -446,15 +446,15 @@ public function findOneHost(int $hostId): ?Host
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = h.host_id
                   AND acl.service_id IS NULL
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 9]
```diff
@@ -505,23 +505,24 @@ public function findOneService(int $hostId, int $serviceId): ?Service
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = h.host_id
                   AND acl.service_id = srv.service_id
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 10]
```diff
@@ -678,15 +680,15 @@ public function findServicesByHost(int $hostId): array
...
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
                   ON acl.host_id = srv.host_id
                   AND acl.service_id = srv.service_id
                 INNER JOIN `:db`.`acl_groups` acg
                   ON acg.acl_group_id = acl.group_id
                   AND acg.acl_group_activate = \'1\'
-                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') '
-            : ' ';
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups). ') ';
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php [hunk 11]
```diff
@@ -573,15 +574,15 @@ public function findServices(): array
...
```
Comment (kduret) [inline src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]:
```suggestion
                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
```

**Comment by kduret** [comment]
if a user is not admin and not linked to an access group, does it break the queries?

File: src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php (hunk 1 above)
Comment (callapa) [reply src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php]: No, but I prefer :)

**Comment by kduret** [comment]
> if a user is not admin and not linked to an access group, does it break the queries?

seen with @callapa it cannot happen :)

**Pull request merged**